src/compiler/checker.ts

```diff
@@ -7521,6 +7558,12 @@ namespace ts {
                 return resolveErrorCall(node);
             }
 
+            // If the expression is of abstract type, then it cannot be instantiated.
+            let valueDecl = expressionType.symbol && expressionType.symbol.valueDeclaration;

```

Comment:
I don't think `.valueDeclaration` will work if you have a module merged with a class:
```ts
declare module MC {
    var v;
}
declare abstract class MC {
    abstract m();
}
new MC(); // expect error here
```